File: examples/evchargers/search/node/searchWithFiltersFastChargersOnly.js
import * as evchargers from "m3o/evchargers";

// Search by giving a coordinate and a max distance, or bounding box and optional filters
async function SearchWithFiltersFastChargersOnly() {
  let evchargersService = new evchargers.EvchargersService(
    process.env.MICRO_API_TOKEN
  );
  let rsp = await evchargersService.search({
    distance: 2000,
    levels: ["3"],
    location: {
      latitude: 51.53336351319885,
      longitude: -0.0252,
    },
    max_results: 2,
  });
  console.log(rsp);
}

await SearchWithFiltersFastChargersOnly();

File: examples/go/evchargers/search/searchWithFiltersFastChargersOnly.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/evchargers"
)

// Search by giving a coordinate and a max distance, or bounding box and optional filters
func SearchWithFiltersFastChargersOnly() {
	evchargersService := evchargers.NewEvchargersService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := evchargersService.Search(&evchargers.SearchRequest{
		Distance: 2000,
		Levels:   []string{"3"},
		Location: &evchargers.Coordinates{
			Latitude:  51.53336351319885,
			Longitude: -0.0252,
		},
	})
	fmt.Println(rsp, err)
}
